### ISSUE TITLE
Govee

### ISSUE THREAD

**Issue by rhoskins61** [LaggAt/hacs-govee#72]
Integration for HA just stopped working. Applied for new API, still causing errors. What's the problem?

**Comment by JohnMachete** [comment]
Same issue, was getting an issue where the Govee object had no property of poll_rate or something, can't find it now since I removed and reinstalled the latest Govee integration and now can't put in my API key, getting 'Cannot connect. Is the API-Key correct and the internet connection working?` error.  Applied for new key and same thing.  Assume either the developer API is down/broken?

**Comment by marcoecheverria** [comment]
Correct. I am having the same problem: Is the API-Key correct and the internet connection working?

**Comment by rhoskins61** [comment]
I sent a quick Govee Support Request.. we'll see what they say. Seems to me to be a Govee issue rather than this integration, but who knows.

**Comment by nitro41992** [comment]
![image](https://user-images.githubusercontent.com/8192635/166973063-9b395a6f-713d-4feb-9b41-6d73a738bce4.png)
Seems like an issue with the API itself, cant even get it to work

![image](https://user-images.githubusercontent.com/8192635/166973203-4dec078d-7e50-4faf-9268-b478b8046511.png)
The documentation says 10,000 per day allowed. The API message says that the max "10000 in 86400000 seconds" which is like 2+ years. Im not sure what that means but I hope we dont have to wait that long for it to be back.

This is definitely a problem with the Govee API. Not sure if we will see a fix soon.

**Comment by trixor** [comment]
Thing stopped working here as well, seems they have an error in their API. Same error as:    Govee API: API: API-Error 429: {"message":"rate limited,the limit is max 10000 in 86400000 seconds. see X-RateLimit-Reset in Response Header"}


The documentation talks about not exceeding 1000 calls per user per day, followed by a maximum a table that shows 10K requests per user per day maximum. 

It seems when they reconfigured the API, they added a few extra zero's to their intended configuration (3 to be exact), as 24h is 86400 seconds instead of what seems to be configured 86400000 seconds

**Comment by schumi2309** [comment]
a new api doesn't work either. I hope the problem will be fixed.

**Comment by zkarampa** [comment]
> the limit is max 10000 in 86400000 seconds.

I guess they meant `ms` rather than `seconds` :facepalm:

**Comment by marcoecheverria** [comment]
I sent a note to api-support@govee.com explaining the situation. I'll keep you posted if I hear from them.

**Comment by Juffle74** [comment]
Same here, mine was set to every 30 secs...

This really need to be a local integration.

**Comment by Scope666** [comment]
Same error:

`2022-05-05 17:51:18 WARNING (MainThread) [govee_api_laggat.govee_api_laggat] error getting state for device GoveeDevice(device='E0:EF:A4:C1:38:5F:xx:xx', model='H6159', device_name='Theater LED', controllable=True, retrievable=True, support_cmds=['turn', 'brightness', 'color', 'colorTem'], support_turn=True, support_brightness=True, support_color=True, support_color_tem=True, online=True, power_state=False, brightness=254, color=(0, 0, 255), color_temp=0, timestamp=1651766818.240145, source=<GoveeSource.API: 'api'>, error='API-Error 429: {"message":"rate limited,the limit is max 10000 in 86400000 seconds. see X-RateLimit-Reset in Response Header"}', lock_set_until=0, lock_get_until=0, learned_set_brightness_max=100, learned_get_brightness_max=100, before_set_brightness_turn_on=False, config_offline_is_off=False): API-Error 429: {"message":"rate limited,the limit is max 10000 in 86400000 seconds. see X-RateLimit-Reset in Response Header"}`

**Comment by omayhemo** [comment]
Seems like the api has changed, but the integration has not.

**Comment by Scope666** [comment]
@trixor explained the problem perfectly above, they added 3 extra zeros to their rate limiting.  You can see it in my log output above.

**Comment by marcoecheverria** [comment]
Here is the response I received from Govee:
Hello ,
Thank you for contacting Govee here.
Due to the limited resources of the API server, the flow of some high-frequency API calls will be limited in order to make the API server more stable and serve more users.
The limit on API call frequency was implemented started May 5, 2022. 
If your current usage scenario calls the API frequently, your application might be affected and may not function properly. 
Therefore, you will need to modify your applications to call the API less frequently. 
Specific rules related to this are listed in the latest API reference document here：
[https://govee-public.s3.amazonaws.com/developer-docs/GoveeDeveloperAPIReference.pdf](https://2h8pfc2d.r.us-east-1.awstrack.me/L0/https:%2F%2Fgovee-public.s3.amazonaws.com%2Fdeveloper-docs%2FGoveeDeveloperAPIReference.pdf/1/0100018094fabbba-0010c6f7-fc6b-4dd5-9927-212443ce638a-000000/kusJX2UqxFjD3A3MrM8YL_RtgRY=269) 


Thank you for your understanding.

Sharon
Govee Customer Support

**Comment by trixor** [comment]
Dear customer,

Thank you for contacting Govee.



Here is the picture attached. X-RateLimit-Limit,X-RateLimit-Remaining,X-RateLimit-Reset (this level of limit works on all APls, which means total requests from oneuser can not be more than 1000 per day).



Your kind understanding is much appreciated.



Best wishes,

Jane


I'm going to respond once more, I assume this is their canned response they give to everyone right now who messages them as they anticipated this to break a lot of people, though right now I think they broke people more than they intended too.

**Comment by a-bianucci** [comment]
So given the error message, and implying years, do we just have to wait 24 hours, what are we thinking here?

**Comment by trixor** [comment]
I've reduced my polling to 600 (from 10), and I'm hoping we'll see a reset in a few hours. I'm not sure when the 24h limit started, did anyone capture this by chance? Reading the documentation, we can actually ask the API how much longer we're in timeout, I may give that a go and post my results.

**Comment by nitro41992** [comment]
@a-bianucci
Seems like I have to wait 3 hours. Probably the same for most here
![image](https://user-images.githubusercontent.com/8192635/167063673-5fbff70d-a8bd-4b24-bfa5-2a2bb621db92.png)

**Comment by a-bianucci** [comment]
@nitro41992 Thanks!  Unfortunately since I just tried, assuming I will have to try again in the morning.

It is strange, I don't think I'm at that volume either. I have 2 lights that are on Govee, and I was configured with "10" as the default.   The lights are auto on\off based on TOD, so essentially 4 calls a day, plus whatever polling the integration does. I can't imagine that I have hit the limit with my use.

I know Govee just did some API change on the 4th, so I'm wondering if there is more to this.

**Comment by a-bianucci** [comment]
> I've reduced my polling to 600 (from 10), and I'm hoping we'll see a reset in a few hours. I'm not sure when the 24h limit started, did anyone capture this by chance? Reading the documentation, we can actually ask the API how much longer we're in timeout, I may give that a go and post my results.

Yeah I can't do that yet.... before I came here I tried to remove, and readd so I can't even get it back in HASS at the moment.  But leveraging Nitro's screenshots, I can see my time is about 2h 45m from now so.... the am it will be :(

**Comment by Bison010** [comment]
I’m having the same issues or rate limit exceeded and I only have 2 govee Light strips triggered by a motion sensor. This just started on the 5th of May. Any help would be appreciated

**Comment by Bison010** [comment]
Mine is back up after updating HASOS to 2022.5.1. Hopefully it stays working

**Comment by rhoskins61** [comment]
Yeah, the integration finally took the new API (that I received yesterday), rebooted HA and all is back.... strange.

**Comment by TomCasavant** [comment]
I didn't have to do anything, mine just started back up sometime last night

**Comment by Morteum9** [comment]
Same here issues here, I have tried disabling the integration and re-enabling. 

Have updated to 2022.5.1 as well this morning, that did not correct it either. 

It was odd that a couple of my lights worked on schedules for wake up routines but not others by the time we got downstairs to start our day. 

Now they are all offline according to Home Assistant.

**Comment by omayhemo** [comment]
Disabled and Re-Enabled the integration this morning and received an initial warning in the logs which said the API is back online.  Is now working fine.

**Comment by JohnMachete** [comment]
Tried reinstalling the integration yesterday and re-inputting my API key but was getting the API error from the integration and couldn't finish adding the integration.  Requested a new API key and had the same issue.  Tried inputting the new API key all day yesterday to no success.  

The new API key now works this morning.  Upped the poll interval to 60, and for 5 devices in 30 minutes I've used about 160 API requests based on the X-RateLimit-Remaining I'm getting back for my API key.  That equals out to about 1 request per devices, per minute, plus me turning them on and off a few times.

**Comment by Morteum9** [comment]
> Disabled and Re-Enabled the integration this morning and received an initial warning in the logs which said the API is back online. Is now working fine.

Tried again just now and not getting the lights to work still... Not vital as the sun is still rising here and the room is light enough from that but certainly going to hear about it if these lights still don't work tonight

**Comment by pgelinas98** [comment]
Mine seems to be running fine still

**Comment by a-bianucci** [comment]
> Tried reinstalling the integration yesterday and re-inputting my API key but was getting the API error from the integration and couldn't finish adding the integration. Requested a new API key and had the same issue. Tried inputting the new API key all day yesterday to no success.
> 
> The new API key now works this morning. Upped the poll interval to 60, and for 5 devices in 30 minutes I've used about 160 API requests based on the X-RateLimit-Remaining I'm getting back for my API key. That equals out to about 1 request per devices, per minute, plus me turning them on and off a few times.

Did the math on mine since I re-enabled it, seeing about 2.4 requests per minute per device.

One thing I did do is run a check on the API to see the reset time, it gave me 24h from the time I ran it vice the first poll earlier.  I wonder if it's a rolling clock of some sort.

**Comment by trixor** [comment]
From Govee support:

```
Dear customer,

I'm sincerely sorry for the mess of the wrong number.

It allows 10k per day and it's 86400 seconds. Thanks for your careful discover. I have conveyed to the relevant department.

Thank you again.
```

Seems the department is aware that they have a bug and are working on allowing 10K requests per day (86400 seconds). My old API key really no longer worked, so I requested a new one. My lights are happy again! Still need to do the math on polling for the number of lights I have in this place.

**Comment by Morteum9** [comment]
I just asked for a new API key and cannot get it to take, it claims "Cannot connect. Is the API-Key correct and the internet connection working?" now...

**Comment by Juffle74** [comment]
Mine is back today and working fine but something seems wrong with all of them showing negative remaining limit and the timestamp...
![Screenshot_20220506-135856_Home Assistant](https://user-images.githubusercontent.com/86912232/167191959-a5f6d03c-3c9f-4ca4-ac5f-1440861a22dc.png)

**Comment by pgelinas98** [comment]
Seems to be having an issue again

**Comment by jsheph** [comment]
Does anyone have a suggestion for calculating what the poll interval should be considering the new rate limitations?
(1000 requests per user per day maximum before triggering a 24 hour timeout)

**Comment by rhoskins61** [comment]
Not working again...

**Comment by trixor** [comment]
I increased to 600 polling, 10-ish lights, no issues so far

**Comment by jsheph** [comment]
> I increased to 600 polling, 10-ish lights, no issues so far

I have about the same qty so I'll give that a shot. T-16 hours till Govee lets my account out of the doghouse to try again.

**Comment by pgelinas98** [comment]
Probably a silly question but how do you adjust the polling?

**Comment by jsheph** [comment]
> Probably a silly question but how do you adjust the polling?

In HA 2022.05.0, go to Devices & Services -> Integrations -> Govee -> Configure -> Polling should be below your API key field.

**Comment by pgelinas98** [comment]
> > Probably a silly question but how do you adjust the polling?
> 
> In HA 2022.05.0, go to Devices & Services -> Integrations -> Govee -> Configure -> Polling should be below your API key field.

Ok thank you. Now I should change the 10 to what?

**Comment by DERFAUSt** [comment]
> > > Probably a silly question but how do you adjust the polling?
> > 
> > 
> > In HA 2022.05.0, go to Devices & Services -> Integrations -> Govee -> Configure -> Polling should be below your API key field.
> 
> Ok thank you. Now I should change the 10 to the 600?

lol....i have the same question...so just change the Poll Interval from 10 to 600?  i just want to be sure so when the timer starts over, it si set right.... .....  and if someone could explain why going from 10 to 600 was the thing that would awesome too :).... ty!

**Comment by Juffle74** [comment]
I might be wrong but I believe the poll interval set the delay between each time the integration check the states of all your govee lights on the govee api server.

So lets say you have 10 govee lights and the poll interval is set at 10 secs.

1. This will result in 86400 hits per day on the 10000 per day rate limit. You will then reach that limit in about 3 hours and wont be able to control any light for the next 21 hours.

2. This will also result in 6 hits on the 10 per minute per device rate limit for each lights.

Setting the poll interval at 600 secs for the same 10 lights will result in 1440 hits on the same limit so you will have 8560 left for turning light on/off, changing color, etc... But doing so will result in HA refreshing state of the govee lights every 10 minutes.

It's all about how many lights you have and a balance between state refreshing and control/change you can do per day.

For my part, I have 7 lights and  I set the poll interval to every 120 secs. That way, I have around 5000 left in the rate limit per day.

Also, one need to make sure that automation don't turn on lights that are already on every time a motion sensor is activated or it will take many hits on the limits.

Anyway that is how I understand how it is working but I might be wrong.

Also, Govee said a while ago that they are working in changing the state request so that it return all lights state at once. This would help.

**Comment by rhoskins61** [comment]
Dumb question for me... how do you start this "timer" over? I changed the interval to 600 and submitted with the current API, which indicates success, however all of my devices are still disabled after reboot.

**Comment by LukeBH** [comment]
> Dumb question for me... how do you start this "timer" over? I changed the interval to 600 and submitted with the current API, which indicates success, however all of my devices are still disabled after reboot.

You need to restart HA Core.

**Comment by rhoskins61** [comment]
That did the trick, thanks.

**Comment by DERFAUSt** [comment]
as of about 9am they are working...i put in my new api...set the poliing to 600.....restarted HA.....boom....gravy :)

ok....  im trying to put my head around this.... formula-wise..

there are 86,400 seconds in a day....  if the poling means how often the integration hits the device in seconds....ie if it is set to 10 then it will poll 86400s/10=8640hits/day....leaving 10000(limit)-8640=1360hits/day from 1 light......?    

so if there are 2 lights....would that change that 8640hits/day to 17284 hits/day?.....sending you 7284 hits over your limit? ..from 2 lights and polling still set to 10..... ?

lol....one more..   

ok....on one light....if setting the polling to 600.... that would be 86400/600=144 hits a day to the api?......leaving 9856 additional hits that day?..... then 10 lights would be that 144x10=1440 hits/day?...leaving 8560 additional hits for the day?....

i appreciate everyone's time.... i hope my maths are right :)

good times math on a saturday morning....there is a song in there somewhere.....

**Comment by trixor** [comment]
@DERFAUSt your assumptions match mine. I've intentionally gone for a really high polling to see if I experience any negatives. Especially given that I plan to expand our Govee usage (that is if the API remains stable). So far I've not found a downside to super high polling but would  be open to hear from all of you.

**Comment by Juffle74** [comment]
The only down side I see is if your turn on or off a light from the Govee app or somewhere else that Home Assistant, il will take up to 10 mins to the change to occur in Home Assistant

**Comment by DERFAUSt** [comment]
well my govee integration worked for 24 hours :)....  so yay! :)....

**Comment by jsheph** [comment]
For those of you who are struggling to find out what polling value to input into the configuration panel, I made a simple calculator that identifies the minimum polling value to use based on the qty of devices (small buffer included). Big thanks to @Juffle74 and @DERFAUSt for your incredible explanations.

Hope this helps. 
[https://github.com/jsheph/Hacs-Govee_Polling_Calculator/blob/88921c38e472fe81eca7b963de5857f3324a571d/Minimum-Value.py](url)

**Comment by DERFAUSt** [comment]
what a bad ass community :).... thank you!.....

**Comment by pgelinas98** [comment]
What is replit? I haven’t heard of it

**Comment by marcoecheverria** [comment]
https://replit.com/. Download it and copy the lines of the code shared by @jsheph  and run it.

**Comment by Bossman5533** [comment]
I have 6 devices and having problems. Very new to all of this. How do you use the calculator?

**Comment by DERFAUSt** [comment]
6 devices....set poling to 70-80....   that should leave you around 3100 additional hits/day.....

**Comment by DERFAUSt** [comment]
> I have 6 devices and having problems. Very new to all of this. How do you use the calculator?

6 devices....set poling to 70-80.... that should leave you around 3100 additional hits/day.....

**Comment by Scope666** [comment]
That's actually GREAT news, a local API would be 1000 times better.  I have most of the devices in my house that are capable running Tasmota.  It's so much nicer not being dependant on the cloud.

**Comment by jmartinha** [comment]
Great news indeed.

**Comment by mikec61** [comment]
All I'm going to say is that a new local API will influence me greatly to buy more of their product.

**Comment by a-bianucci** [comment]
Second that.  It’s the only reason I haven’t TBH is local control.

Fine for the kids room that has a few on odds a day vice something that would have more.  They have some other products that are nice.

I do wish we had RGBIC control or access to scenes.


Aaron Bianucci

On May 11, 2022, at 20:53, mikec61 ***@***.***> wrote:

﻿

All I'm going to say is that a new local API will influence me greatly to buy more of their product.

**Comment by Normanras** [comment]
Would it help if we all email support@govee expressing our desire for a local API configuration? I think that's great news indeed and I'm wondering if increased pressure from us will get them to push it through the roadmap a bit more quickly. I doubt they are searching Github issues for the breadth of use, though I'm sure they see how many API calls they are getting.

**Comment by Juffle74** [comment]
Already done a while ago while talking with the tech support within the Govee app... I might do it again by email this time.

Dont forget to ask for full control of the lights. Like on/off, brightness, scenes, mode, etc...

**Comment by adamwmitchell** [comment]
Just checking to see if anything changed in the past two days....I finally had time to get into HA and check on why my Govee lights weren't working and came across this page. Yesterday I updated HA. I removed Govee from Hacs and re-downloaded the latest build. I also received a new API key and I am still getting the error message about the API key not being correct or is the internet connected. I am attempting to put in the API Key and an interval of 600.  Can anyone tell me where to go to see my Govee Api stats....I am not sure when my 24hrs starts again.

**Comment by adamwmitchell** [comment]
I was just too impatient....my 24 hours must have not been up yet. It just now worked. However, I would still love to know where I can go and get those stats for my Govee Api...Please and Thank you!

**Comment by CyberCowboy** [comment]
> I was just too impatient....my 24 hours must have not been up yet. It just now worked. However, I would still love to know where I can go and get those stats for my Govee Api...Please and Thank you!

Did you ever figure out how to find out your reset time?

**Comment by Normanras** [comment]
Has anyone here received an update from Govee on this? I'm still getting API issues, so my guess it still isn't implemented. Last I heard from their support team was my comment here on May 13 saying that their intention was to implement something. Anyone hear anything? I have multiple switches that would be put to good use that are just sitting around...

**Comment by TomCasavant** [comment]
I'm still getting the occasional issue as well after changing the polling time. I've got an unusually high number of govee bulbs though, so I might just have to switch a couple out for now

**Comment by pslavickas** [comment]
I just bought about 4-5 new Govee devices and am having the same problem. It certainly seems a bit odd for Govee to punish users for purchasing MORE of their devices. This absolutely doesn't seem to scale well at all.

**Comment by Normanras** [comment]
Was this done in the Govee mobile app?

**Comment by TomCasavant** [comment]
> 

https://app-h5.govee.com/user-manual/wlan-guide

**Comment by Juffle74** [comment]
Sorry, I mean "Govee LAN Control" as a Home Assistant integration available in HACS.

https://github.com/wez/govee-lan-hass
